second_sql_assignment.sql

```diff
@@ -0,0 +1,232 @@
+
+
+SELECT * FROM Employees
+WHERE LEFT(LastName,1) IN ('B','R','S');
+
+-- RETRIEVE ALL EMPLOYEES WHOSE FIRST NAME BEGINS WITH ANY LETTER FROM 'A'THROUGH 'M'.
+
+SELECT * FROM Employees
+WHERE LEFT(FirstName,1) BETWEEN 'A' AND 'M';
+
+-- LIST EMPLOYEES WHOSE LAST NAME DOESN'T START WITH A VOWEL.
+
+SELECT * FROM Employees
+WHERE LEFT(LastName,1) NOT IN ('A','E','I','O','U');
+
+-- IDENTIFY EMPLOYEES EARNING MORE THAN $80,000 ANNUALLY.
+
+SELECT * FROM Employees
+WHERE Salary>80000;
+
+-- FIND EMPLOYEES WHO JOINED THE COMPANY BEFOR 2020.
+
+SELECT * FROM Employees
+WHERE HireDate < '2020-01-01';
+
+-- LIST ALL EMPLOYEES NOT NAMED 'JOHN' IN FIRST NAME.
+
+SELECT * FROM Employees
+WHERE FirstName != 'John';
+
+-- IDENTIFY MARKETING DEPARTMENT EMPLOYEES EARNING $60,000 OR LESS WHO WERE HORED AFTER JUNE 30,2019
+
+SELECT * FROM Employees
+WHERE Department ='Marketing'
+	  AND Salary <= 60000
+	  AND HireDate > '2019-06-30';
+
+-- FIND EMPLOYEE WHOSE FIRST NAME CONTAINS THE LETTERS 'AN' ANYWHERE AND ENDS WITH 'E'.
+
+SELECT * FROM Employees
+WHERE Firstname LIKE '%an%e';
+
+
+QUESTION 2
+
+
+CREATE TABLE Sales(
+SaleID INT PRIMARY KEY,
+ProductID INT,
+CustomerID INT,
+SaleDate DATE,
+Quantity INT,
+UnitPrice DECIMAL(10,2));
+
+
+
+INSERT INTO Sales (SaleID, ProductID, CustomerID, SaleDate, Quantity, UnitPrice) VALUES
+(1, 101, 201, '2024-01-15', 2, 15.99),
+(2, 102, 202, '2024-01-16', 1, 29.50),
+(3, 103, 203, '2024-01-17', 5, 9.99),
+(4, 104, 204, '2024-01-18', 3, 20.00),
+(5, 105, 205, '2024-01-19', 1, 45.00),
+(6, 101, 206, '2024-02-01', 4, 15.99),
+(7, 102, 207, '2024-02-02', 2, 29.50),
+(8, 103, 208, '2024-02-03', 3, 9.99),
+(9, 104, 209, '2024-02-04', 1, 20.00),
+(10, 105, 210, '2024-02-05', 6, 45.00);
+
+CREATE TABLE Products (
+    ProductID INT PRIMARY KEY,
+    ProductName VARCHAR(100),
+    Category VARCHAR(50));
+
+
+INSERT INTO Products (ProductID, ProductName, Category) VALUES
+(101, 'Wireless Mouse', 'Electronics'),
+(102, 'Mechanical Keyboard', 'Electronics'),
+(103, 'Gaming Headset', 'Electronics'),
+(104, 'Office Chair', 'Furniture'),
+(105, 'Desk Lamp', 'Furniture');
+
+
+CREATE TABLE Customers (
+    CustomerID INT PRIMARY KEY,
+    CustomerName VARCHAR(100),
+    City VARCHAR(50),
+    State VARCHAR(2));
+
+
+INSERT INTO Customers (CustomerID, CustomerName, City, State) VALUES
+(201, 'John Doe', 'New York', 'NY'),
+(202, 'Jane Smith', 'Los Angeles', 'CA'),
+(203, 'Sam Brown', 'Chicago', 'IL'),
+(204, 'Emily White', 'Houston', 'TX'),
+(205, 'Michael Green', 'Phoenix', 'AZ');
+
+-- CALCULATE THE TOTAL SALES AMOUNT FOR EACH PRODUCT. DISPLAY THE PRODUCT ID AND TOTAL SALES AMOUNT.
+
+SELECT ProductID, SUM(quantity * UnitPrice) as Total_Sales_Amount
+FROM Sales
+GROUP BY ProductID;
+
+-- FIND THE AVERAGE QUANTITY SOLD PER SALE FOR EACH PRODUCT CATEGORY.
+
+SELECT p.Category, AVG(s.Quantity) AS AverageQuantitySold
+FROM Sales s
+JOIN Products p ON s.ProductID = p.ProductID
```

Comment:
What if I have products that are not sold, What will be the output of this query?

Reply:
Since JOIN (INNER JOIN) is used, only values common in both tables will be displayed.